components/menu/index.css

```diff
@@ -10,170 +10,676 @@ OF ANY KIND, either express or implied. See the License for the specific languag
   }
 }
 
+.spectrum-Menu-itemIcon {
+  fill: var(--highcontrast-menu-item-label-icon-color-default, var(--mod-menu-item-label-icon-color-default, var(--spectrum-menu-item-label-icon-color-default)));
+}
+
 .spectrum-Menu-checkmark {
+  align-self: center;
+  fill: var(--highcontrast-menu-checkmark-icon-color-default, var(--mod-menu-checkmark-icon-color-default, var(--spectrum-menu-checkmark-icon-color-default)));
   transform: scale(1);
   opacity: 1;
 }
 
 .spectrum-Menu-item {
   cursor: pointer;
   position: relative;
-  display: flex;
   align-items: center;
 
   box-sizing: border-box;
 
-  padding: var(--spectrum-listitem-texticon-padding-y)
-    var(--spectrum-listitem-texticon-padding-right)
-    var(--spectrum-listitem-texticon-padding-y)
-    var(--spectrum-listitem-texticon-padding-left);
+  background-color: var(--mod-menu-item-background-color-default, var(--spectrum-menu-item-background-color-default));
+
+  border-inline-start: var(--mod-menu-item-focus-indicator-width, var(--spectrum-menu-item-focus-indicator-width)) solid transparent;
+  
+  line-height: var(--mod-menu-item-label-line-height, var(--spectrum-menu-item-label-line-height));
 
+  min-block-size: var(--mod-menu-item-min-height, var(--spectrum-menu-item-min-height));
+  padding-block-start: var(--mod-menu-item-top-edge-to-text, var(--spectrum-menu-item-top-edge-to-text));
+  padding-block-end: var(--mod-menu-item-bottom-edge-to-text, var(--spectrum-menu-item-bottom-edge-to-text));
+  padding-inline: var(--mod-menu-item-label-inline-edge-to-content, var(--spectrum-menu-item-label-inline-edge-to-content));
+  
   margin: 0;
+  text-decoration: none;
+}
 
-  border-inline-start: var(--spectrum-listitem-texticon-focus-indicator-size) solid
-    transparent;
+.spectrum-Menu-item {
+  display: grid;
+  grid-template-areas:
+    "selectedArea chevronAreaCollapsible checkmarkArea iconArea labelArea        valueArea actionsArea chevronAreaDrillIn"
+    ".            .                      .             .        descriptionArea  .         .           .                 "
+    ".            .                      .             .        submenuArea      .         .           .                 ";
+  
+  grid-template-columns: auto auto auto auto 1fr auto auto auto;
+  grid-template-rows: 1fr auto;
+}
 
-  min-block-size: var(--spectrum-listitem-texticon-height);
+.spectrum-Menu-item .spectrum-Menu-item {
+  display: grid;
+  grid-template-areas: "submenuItemLabelArea submenuItemValueArea";
+  grid-template-columns: auto auto;
+  grid-template-rows: 1fr auto;
+}
 
-  font-size: var(--spectrum-listitem-texticon-text-size);
-  font-weight: var(--spectrum-listitem-texticon-text-font-weight);
-  font-style: normal;
-  text-decoration: none;
+.spectrum-Menu-item .spectrum-Menu-item .spectrum-Menu-itemLabel {
+  grid-area: submenuItemLabelArea;
+}
+
+.spectrum-Menu-item .spectrum-Menu-item .spectrum-Menu-itemValue {
+  grid-area: submenuItemValueArea;
+}
+
+.spectrum-Menu-item:focus {
```

Comment:
Lines 388-418 appear to be redundant with lines 420-452. Suggest adding .spectrum-Men-item:focus to line 420.

But I'd check with engineering to make sure that controlling :focus in the CSS won't interfere with their work. @GarthDB can you chime in on this? Should we keep :focus on menu items or remove it?

Reply:
Yeah those were almost completely redundant. I consolidated.